From: Christoph Hellwig <hch@infradead.org>
Date: Mon, 17 Oct 2011 13:56:50 -0400
Subject: [PATCH] target: remove the TRANSPORT_REMOVE state
Git-commit: bfaf40ada2e15bc972cab4cd5452a88720e30647
References: FATE#313550
Patch-Mainline: v3.2

We never queue an command with this state, and only set it in a completely
bogus place in tcm_fc.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Nicholas Bellinger <nab@linux-iscsi.org>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/target/target_core_transport.c |    3 ---
 drivers/target/tcm_fc/tfc_cmd.c        |    1 -
 include/target/target_core_base.h      |    1 -
 3 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/drivers/target/target_core_transport.c b/drivers/target/target_core_transport.c
index cdfbc49..ca2cc6a 100644
--- a/drivers/target/target_core_transport.c
+++ b/drivers/target/target_core_transport.c
@@ -4811,9 +4811,6 @@ get_cmd:
 		case TRANSPORT_COMPLETE_OK:
 			transport_generic_complete_ok(cmd);
 			break;
-		case TRANSPORT_REMOVE:
-			transport_put_cmd(cmd);
-			break;
 		case TRANSPORT_FREE_CMD_INTR:
 			transport_generic_free_cmd(cmd, 0);
 			break;
diff --git a/drivers/target/tcm_fc/tfc_cmd.c b/drivers/target/tcm_fc/tfc_cmd.c
index 446008b..863f41c 100644
--- a/drivers/target/tcm_fc/tfc_cmd.c
+++ b/drivers/target/tcm_fc/tfc_cmd.c
@@ -268,7 +268,6 @@ static void ft_recv_seq(struct fc_seq *sp, struct fc_frame *fp, void *arg)
 
 	if (IS_ERR(fp)) {
 		/* XXX need to find cmd if queued */
-		cmd->se_cmd.t_state = TRANSPORT_REMOVE;
 		cmd->seq = NULL;
 		transport_generic_free_cmd(&cmd->se_cmd, 0);
 		return;
diff --git a/include/target/target_core_base.h b/include/target/target_core_base.h
index 9f99f53..4bb7411 100644
--- a/include/target/target_core_base.h
+++ b/include/target/target_core_base.h
@@ -96,7 +96,6 @@ enum transport_state_table {
 	TRANSPORT_ISTATE_PROCESSING = 11,
 	TRANSPORT_ISTATE_PROCESSED = 12,
 	TRANSPORT_KILL		= 13,
-	TRANSPORT_REMOVE	= 14,
 	TRANSPORT_FREE		= 15,
 	TRANSPORT_NEW_CMD_MAP	= 16,
 	TRANSPORT_FREE_CMD_INTR = 17,
-- 
1.7.4.2

